src/components/services/WalletConnectService.tsx

```diff
@@ -7,47 +7,13 @@ import { EdgeAccount } from 'edge-core-js'
 import * as React from 'react'
 
 import { ENV } from '../../env'
-import { useMount } from '../../hooks/useMount'
+import { useAsyncEffect } from '../../hooks/useAsyncEffect'
 import { getAccounts, getWalletIdFromSessionNamespace } from '../../hooks/useWalletConnect'
 import { useWatch } from '../../hooks/useWatch'
 import { WcSmartContractModal } from '../modals/WcSmartContractModal'
 import { Airship } from '../services/AirshipInstance'
 
-let walletConnectRef: Web3Wallet | undefined
-
-export const walletConnectPromise: Promise<Web3Wallet> = new Promise((resolve, reject) => {
```

Comment:
I think the `getClient` method should live in this file, and should be async.

Then we can implement `getClient` as follows:

```ts
const getClient = async () => {
  if (walletConnectClient != null) return walletConnectClient
  return new Promise(resolve => waitingClients.push(resolve))
}

const waitingClients: Array<(client: Web3Wallet) => void> = []
```

Basically, if we have a client, return it. If we don't have a client, add a callback to the list and wait around for it to show up.

Then in your `useAsyncEffect`, you can do `waitingClients.forEach(f => f(walletConnectClient))` to un-block all the waiting code.